js-library/src/services/ens.ts

```diff
@@ -0,0 +1,176 @@
+import { utils, Contract, Signer, providers } from 'ethers'
+import { NULL_ADDRESS } from '../constants/constants'
+import ENSRegistryContract from '../contracts/ENSRegistry.sol/ENSRegistry.json'
+import PublicResolverContract from '../contracts/PublicResolver.sol/PublicResolver.json'
+import SubdomainRegistrarContract from '../contracts/SubdomainRegistrar.sol/SubdomainRegistrar.json'
+import { waitTransaction } from '../utils/tx'
+import { joinPublicKey, isPublicKeyValid, splitPublicKey } from '../utils/keys'
+
+const { keccak256, toUtf8Bytes, namehash } = utils
+
+/**
+ * Contract addresses
+ */
+export const ENS_REGISTRY_ADDRESS = process.env.ENS_REGISTRY_ADDRESS
+export const PUBLIC_RESOLVER_ADDRESS = process.env.PUBLIC_RESOLVER_ADDRESS
+export const SUBDOMAIN_REGISTRAR_ADDRESS = process.env.SUBDOMAIN_REGISTRAR_ADDRESS
+
+/**
+ * Contract metadata
+ */
+export const ENSRegistryMeta = ENSRegistryContract
+export const PublicResolverMeta = PublicResolverContract
+export const SubdomainRegistrarMeta = SubdomainRegistrarContract
+
+/**
+ * ENS Class
+ * Provides interface for interaction with the ENS smart contracts
+ */
+export class ENS {
+  private _provider: providers.JsonRpcProvider
+  private ensRegistryContract: Contract
+  private subdomainRegistrarContract: Contract
+  private publicResolverContract: Contract
```

Comment:
let's keep the conventional underscore prefix at the other private fields as well